File: Huddle.MetricWebApp/Models/Reason.cs
using Huddle.MetricWebApp.Models;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace Huddle.MetricWebApp.Models
{
    public enum TrackingFrequency
    {
        Daily,
        Weekly
    }
    public class Reason:IQueryResult
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public Metric Metric{ get; set; }

        public DateTime StartDate { get; set; }
        public string ReasonTracking { get; set; }
        public TrackingFrequency? TrackingFrequency { get; set; }

        public string ValueType { get; set; }
        public int State { get; set; }
    }
}

File: Huddle.MetricWebApp/app/shared/models/reason.ts
import { Metric} from './metric';
import { State } from './state';
import { QueryResult } from './queryResult';

export class Reason implements QueryResult{
    id?: number;
    name?: string;
    metric?: Metric;
    reasonState?: State;
    startDate?: Date;
    isEditable?: boolean;
    reasonTracking?: string;
    valueType: string;
    trackingFrequency: string;
   
}